cli/processexecutor.cpp

```diff
@@ -239,7 +241,7 @@ bool ProcessExecutor::handleRead(int rpipe, unsigned int &result, const std::str
         if (!buf.empty()) {
             // TODO: avoid string splitting
             auto parts = splitString(buf, ';');
-            if (parts.size() != 3)
+            if (parts.size() != 4)
```

Comment:
what happens if the extra comment contains semicolons. if a quick and easy solution is to remove semicolons from comments then feel free to do that. we could also consider using some other separator than semicolon.